bolparser.py

```diff
@@ -0,0 +1,118 @@
+__author__ = 'Amogh Jalihal'
+
+import json
+import re
+import sys
+
+class bolParser:
+    def __init__(self, bol):
+        self.bol = bol
+        self.Avartan = list()
+        self.Vibhaag = list()
+        self.Matra = list()
+        self.num_avartans = 0
+        self.taaldict = {}
+        
+        self.__cleanupBol()
+        self.__getTaalDict()
+        self.__getAvartan()
+        self.__getVibhaag()
+        self.__getMatra()
+        
+        self.num_avartans = len(self.Avartan)
+
+    def __cleanupBol(self):
+        # Replace multiple spaces in input with single space
+        self.bol = " ".join(self.bol.split())
+        # Remove padded | if any
+        self.bol = re.sub("[ ]*\|[ ]*","|",self.bol)
+
+    def __getTaalDict(self):
+        with open("data/taals.json",'r') as f:
+            self.taaldict = json.load(f)
+        
+    def __getAvartan(self):
+        self.Avartan = self.bol.split('||')[:-1]
+
+    def __getVibhaag(self):
+        self.Vibhaag = [av.split('|') for av in self.Avartan]
+
+    def __getMatra(self):
+        GroupedAkshars = [[v.split(' ') for v in V] for V in self.Vibhaag]
+        for av in GroupedAkshars:
+            for v in av:
+                for a in v:
+                    self.Matra.append(a)
+        
+    def tokenize(self,matra,matraDensity):
+        tokensCompound = ['tirakita','tita']
+        tokensSimple = ['dha','dhi','ghe','tun','na','ti','ta','ka','S']
+        
+        alltokens = tokensSimple + tokensCompound
+        if len(matra) == 0:
+            return(matraDensity)
+        else:
+            for C in tokensCompound:
+                loc = matra.find(C)
+                if loc != -1:
+                    matra = matra.replace(C,'',1)
+                    if C == 'tirakita':
+                        matraDensity.append(4)
+                    if C == 'tita':
+                        matraDensity.append(2)
+                    if len(matra) == 0:
+                        return(matraDensity)
+            count = 0
+            while len(matra) > 0:
+                valid = False
+                for cs in alltokens:
+                    if cs in matra:
+                        valid = True
+                if not valid:
+                    print("Unidentified akshar!\n Dump:")
+                    print(matra)
+                    print("Exiting..")
+                    sys.exit()
```

Comment:
Better to avoid `sys.exit` as we need the main REPL running regardless. 
This should probably `pass` and wait for the next bol/input. 
Maybe call a reset method so that all variables in this class get flushed.

Reply:
Good idea. I'm not familiar with the `code` package. Can you point me to  some documentation on the correct way to handle an exception?

Reply:
I think raising a `ValueError` should be good. This will also ensure the user doesn't get kicked out of the REPL.
http://hplgit.github.io/primer.html/doc/pub/input/._input-bootstrap007.html